src/main/java/command/DeleteCommand.java

```diff
@@ -0,0 +1,50 @@
+package command;
+
+import fridayException.FridayException;
+import fridayException.InvalidDeleteArgument;
+import storage.Storage;
+import task.Task;
+import task.TaskList;
+import ui.Ui;
+
+import java.io.IOException;
+
+/**
+ * Represents a command to delete a task from the task list.
+ */
+public class DeleteCommand extends Command {
+    private final int taskIndex;
+
+    /**
+     * Constructs a DeleteCommand with the specified task index.
+     *
+     * @param taskIndex The index of the task to be deleted from the task list.
+     */
+    public DeleteCommand(int taskIndex) {
+        this.taskIndex = taskIndex;
+    }
+
+    /**
+     * Deletes the task from the task list, displays a message to the user, and saves the task list to file.
+     *
+     * @param tasks The task list to be modified by the command.
+     * @param ui The user interface to display messages to the user.
+     * @param storage The storage to save the task list to file.
+     * @throws FridayException If an error occurs during execution of the command.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws FridayException {
+        try {
+            if (taskIndex < 0 || taskIndex >= tasks.size()) {
+                throw new InvalidDeleteArgument();
+            }
+            Task taskToDelete = tasks.getTask(taskIndex);  // Get the task to be deleted
+            tasks.removeTask(taskIndex);  // Remove the task from the list
+            ui.showTaskRemoved(taskToDelete, tasks.size());  // Show the user that the task was removed
+
+            storage.save(tasks.getTasks());  // Save the updated task list to the file
```

Comment:
I like the use of trailing comments to explain what is happening at those lines, it makes following the code much easier 👍